lint/rule.go

```diff
@@ -14,10 +14,295 @@
 
 package lint
 
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/jhump/protoreflect/desc"
+)
+
 // Rule defines a lint rule that checks Google Protobuf APIs.
 type Rule interface {
-	// Info returns metadata about a rule.
-	Info() RuleInfo
-	// Lint performs the linting process.
-	Lint(Request) ([]Problem, error)
+	// GetName returns the name of the rule.
+	GetName() RuleName
```

Comment:
Please follow Go idioms: https://golang.org/doc/effective_go.html#Getters -- use `Name()` instead.

Reply:
Since rule authors will be interacting with the protoreflect library that uses `GetFoo`, local consistency is more important.

Reply:
protoreflect is a special case due to their decision to maintain consistence with protobuf. But in our godoc, we will not see any `GetFoo` from the protoreflect library. So I think we should keep Go idioms.

Reply:
But rule authors are likely to use `GetMessageTypes`, `GetFields`, etc., and should not have to think about whether this particular item they are using has a `Get-` prefix.

Also, it would require `MessageRule`, `MethodRule`, etc., to adopt different field names in their structs.

Therefore, this should stay as-is.

Reply:
Okay, conflict of field and function names is a legit excuse.